sdk/src/test/java/com/mapbox/maps/ViewAnnotationManagerAddTest.kt

```diff
@@ -0,0 +1,228 @@
+
+  private lateinit var viewAnnotationManager: ViewAnnotationManagerImpl
+  private lateinit var expectedView: View
+  private lateinit var frameLayoutParams: FrameLayout.LayoutParams
+
+  @Before
+  fun setUp() {
+    every { mapView.getMapboxMap() } returns mapboxMap
+    every { mapView.mapController.pluginRegistry.viewPlugins } returns mutableMapOf()
+    viewAnnotationManager = ViewAnnotationManagerImpl(mapView)
+    expectedView = mockk(relaxed = true)
+    frameLayoutParams = mockk()
+    frameLayoutParams.width = layoutParamsWidthActual
+    frameLayoutParams.height = layoutParamsHeightActual
+    every { mapView.context } returns mockk()
+    every { expectedView.layoutParams } returns frameLayoutParams
+    every { mapboxMap.addViewAnnotation(any(), any()) } returns ExpectedFactory.createNone()
+  }
+
+  @After
+  fun tearDown() {
+    every { mapboxMap.removeViewAnnotation(any()) } returns ExpectedFactory.createNone()
+    viewAnnotationManager.destroy()
+  }
+
+  @Test
+  fun addViewAnnotationWithSyncResId() {
+    mockkStatic(LayoutInflater::class)
+    every { LayoutInflater.from(any()).inflate(resIdActual, mapView, false) } returns expectedView
+    try {
+      val actualView = viewAnnotationManager.addViewAnnotation(
+        resIdActual,
+        viewAnnotationOptionsActual
+      )
+      Assert.assertEquals(runtimeExceptionThrown, false)
+      Assert.assertEquals(expectedView, actualView)
+      verify(exactly = 1) {
+        mapboxMap.addViewAnnotation(
+          any(),
+          viewAnnotationOptionsExpected
+        )
+      }
+    } catch (e: RuntimeException) {
+      Assert.assertEquals(runtimeExceptionThrown, true)
+    } finally {
+      unmockkStatic(LayoutInflater::class)
+    }
+  }
+
+  @Test
+  fun addViewAnnotationWithAsyncResId() {
+    val asyncInflater = mockk<AsyncLayoutInflater>()
+    val callback = slot<AsyncLayoutInflater.OnInflateFinishedListener>()
+    every { asyncInflater.inflate(resIdActual, mapView, capture(callback)) } answers {
+      callback.captured.onInflateFinished(expectedView, resIdActual, mapView)
+    }
+    try {
+      viewAnnotationManager.addViewAnnotation(
+        resIdActual,
+        viewAnnotationOptionsActual,
+        asyncInflater,
+      ) {
+        Assert.assertEquals(runtimeExceptionThrown, false)
+        Assert.assertEquals(expectedView, it)
+        verify(exactly = 1) {
+          mapboxMap.addViewAnnotation(
+            any(),
+            viewAnnotationOptionsExpected
+          )
+        }
+      }
+    } catch (e: RuntimeException) {
+      Assert.assertEquals(runtimeExceptionThrown, true)
+    }
+  }
+
+  @Test
+  fun addViewAnnotationWithView() {
+    try {
+      val actualView = mockk<View>(relaxed = true)
+      every { actualView.layoutParams } returns frameLayoutParams
+      viewAnnotationManager.addViewAnnotation(
+        actualView,
+        viewAnnotationOptionsActual
+      )
+      Assert.assertEquals(runtimeExceptionThrown, false)
```

Comment:
not sure that you need handle such cases in parametrized tests
maybe move test with exception to `ViewAnnotationManagerTest`?

Reply:
Why not? it's common error for all 3 addition functions so fits good to parameterized imho. One specific runtime exception for adding already inflated view is indeed handled separately.

Reply:
okay, let's leave as is
but if it's not an exception check, `assertEquals(runtimeExceptionThrown, false)` inside try-catch block useless